android/src/main/kotlin/com/airship/flutter/FlutterAutopilot.kt

```diff
@@ -12,34 +12,59 @@ import com.urbanairship.push.*
 import com.urbanairship.push.notifications.AirshipNotificationProvider
 import com.urbanairship.push.notifications.NotificationArguments
 import androidx.annotation.XmlRes
+import com.airship.flutter.AirshipBackgroundExecutor.Companion.BackgroundMessageResult.*
+import com.airship.flutter.AirshipBackgroundExecutor.Companion.handleBackgroundMessage
 import com.urbanairship.AirshipConfigOptions
 import com.urbanairship.UAirship.getApplicationContext
 import com.urbanairship.analytics.Analytics
 import com.urbanairship.channel.AirshipChannelListener
 import com.urbanairship.preferencecenter.PreferenceCenter
-import kotlinx.coroutines.CoroutineScope
+import com.urbanairship.push.notifications.NotificationResult
 import kotlinx.coroutines.Dispatchers
 import kotlinx.coroutines.flow.first
-import kotlinx.coroutines.flow.single
-import kotlinx.coroutines.flow.singleOrNull
 import kotlinx.coroutines.runBlocking
 
 const val PUSH_MESSAGE_BUNDLE_EXTRA = "com.urbanairship.push_bundle"
 
 class FlutterAutopilot : Autopilot() {
     var config: AirshipConfigOptions? = null
 
+    private val appContext: Context
+        get() = getApplicationContext()
+
     override fun onAirshipReady(airship: UAirship) {
         super.onAirshipReady(airship)
 
         Log.i("FlutterAutopilot", "onAirshipReady")
 
+        // If running in the background, start the background Isolate
+        // so that we can communicate with the Flutter app.
+        if (!appContext.isAppInForeground()) {
+            AirshipBackgroundExecutor.startIsolate(appContext)
+        }
+
         // Register a listener for inbox update event
         MessageCenter.shared().inbox.addListener {
             EventManager.shared.notifyEvent(InboxUpdatedEvent())
         }
 
-        airship.pushManager.notificationProvider = object : AirshipNotificationProvider(UAirship.getApplicationContext(), airship.airshipConfigOptions) {
+        airship.pushManager.notificationProvider = object : AirshipNotificationProvider(appContext, airship.airshipConfigOptions) {
+            override fun onCreateNotification(
+                context: Context,
+                arguments: NotificationArguments
+            ): NotificationResult {
+                if (appContext.isAppInForeground()) {
```

Comment:
I think generally people will want the opposite of what we providing here. In the foreground they might want to do something special, but if they are in the background show the notification.

I am worried we are over complicating this and should just tell them we received a push in the background to do custom processing.